installinstallmacos.py

```diff
@@ -30,42 +30,110 @@
 import plistlib
 import subprocess
 import sys
-import time
 import urlparse
 import xattr
 from xml.dom import minidom
 from xml.parsers.expat import ExpatError
+from distutils.version import LooseVersion
 
 
-DEFAULT_SUCATALOG = (
-    'https://swscan.apple.com/content/catalogs/others/'
-    'index-10.13seed-10.13-10.12-10.11-10.10-10.9'
-    '-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog.gz')
+DEFAULT_SUCATALOGS = {
+    '17': 'https://swscan.apple.com/content/catalogs/others/'
+          'index-10.13-10.12-10.11-10.10-10.9'
+          '-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog',
+    '18': 'https://swscan.apple.com/content/catalogs/others/'
+          'index-10.14-10.13-10.12-10.11-10.10-10.9'
+          '-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog',
+}
+
 
 SEED_CATALOGS_PLIST = (
     '/System/Library/PrivateFrameworks/Seeding.framework/Versions/Current/'
     'Resources/SeedCatalogs.plist'
 )
 
 
+def get_board_id():
+    '''Gets the local system board ID'''
+    ioreg_cmd = ['ioreg', '-p', 'IODeviceTree', '-r', '-n', '/', '-d', '1']
+    try:
+        ioreg_output = subprocess.check_output(ioreg_cmd).splitlines()
+        for line in ioreg_output:
+            if 'board-id' in line:
+                board_id = line.split(" ")[-1]
+                board_id = board_id[board_id.find('<"')+2:board_id.find('">')]
+                return board_id
+    except subprocess.CalledProcessError, err:
+        raise ReplicationError(err)
+
+
+def is_a_vm():
+    '''Determines if the script is being run in a virtual machine'''
+    sysctl_cmd = ['/usr/sbin/sysctl', 'machdep.cpu.features']
+    try:
+        sysctl_output = subprocess.check_output(sysctl_cmd)
+        cpu_features = sysctl_output.split(" ")
+        is_vm = False
+        for i in range(len(cpu_features)):
+            if cpu_features[i] == "VMM":
+                is_vm = True
+    except subprocess.CalledProcessError, err:
+        raise ReplicationError(err)
+    return is_vm
+
+
+def get_hw_model():
+    '''Gets the local system ModelIdentifier'''
+    sysctl_cmd = ['/usr/sbin/sysctl', 'hw.model']
+    try:
+        sysctl_output = subprocess.check_output(sysctl_cmd)
+        hw_model = sysctl_output.split(" ")[-1].split("\n")[0]
+    except subprocess.CalledProcessError, err:
+        raise ReplicationError(err)
+    return hw_model
+
+
+def get_current_build_info():
+    '''Gets the local system build'''
+    build_info = []
+    sw_vers_cmd = ['/usr/bin/sw_vers']
+    try:
+        sw_vers_output = subprocess.check_output(sw_vers_cmd).splitlines()
+        for line in sw_vers_output:
+            if 'ProductVersion' in line:
+                build_info.insert(0, line.split("\t")[-1])
+            if 'BuildVersion' in line:
+                build_info.insert(1, line.split("\t")[-1])
+    except subprocess.CalledProcessError, err:
+        raise ReplicationError(err)
```

Comment:
ReplicationError seems the wrong error to raise -- this has nothing to do with replicating the files from Apple

Reply:
Changed to a simpler output